### PR TITLE
feat(Core/SavingSystem): remove old player saving system

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Remove old player saving system
- Add `TaskScheduler` for save players before shutdown

## Issues Addressed:
- Part #9630

## Tests Performed:
- in game

## How to Test the Changes:
1. Just check save players

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by temperrr** [comment]
Tested

- [x] logout save 
- [x] .save command 
- [x] intermittent save (like 3 minutes)
- [ ] .server restart save not working!

**Comment by Winfidonarleyan** [comment]
> Tested
> 
> * [x]  logout save
> * [x]  .save command
> * [x]  intermittent save (like 3 minutes)
> * [ ]  .server restart save not working!

Thanks

**Comment by Winfidonarleyan** [comment]
@temperrr i added players save system before shutdown server

File: src/server/game/World/World.cpp [hunk 1]
```diff
@@ -2705,6 +2710,33 @@ void World::ShutdownServ(uint32 time, uint32 options, uint8 exitcode, const std:
     m_ShutdownMask = options;
     m_ExitCode = exitcode;
 
+    auto const& playersOnline = GetActiveSessionCount();
+    
+    if (time < 5 && playersOnline)
```
Comment (Winfidonarleyan) [inline src/server/game/World/World.cpp:2715]:
```suggestion
    auto const& playersOnline = GetActiveSessionCount();

    if (time < 5 && playersOnline)
```

**Comment by temperrr** [comment]
- [x] server restart 5
- [x] logout save
- [x] .save command
- [x] intermittent save

**Review by Nyeriah — APPROVED** [review]

**Pull request merged**